src/components/PopupWithSubmit.js

```diff
@@ -0,0 +1,33 @@
+import Popup from "./Popup.js";
+
+class PopupWithSubmit extends Popup {
+  constructor(popupSelector) {
+    super({ popupSelector });
+    this._popupForm = this._popupElement.querySelector(".modal__form");
+    this._submitButton = this._popupForm.querySelector(".modal__button");
+    this._submitButtonText = this._submitButton.textContent;
+  }
+
+  setSubmitAction(handleFormSubmit) {
```

Comment:
Great job implementing confirmation popup and the functionality related to it! In general, it can support any confirmation.